src/lib.rs

```diff
@@ -67,11 +65,23 @@ impl Default for Client {
         // TODO: make configurable in `Client::builder()`.
         connector.set_keepalive(Some(TCP_KEEPALIVE));
 
-        #[cfg(feature = "tls")]
-        let connector = HttpsConnector::new_with_connector({
-            connector.enforce_http(false);
-            connector
-        });
+        #[cfg(any(feature = "tls", feature = "rustls"))]
+        connector.enforce_http(false);
+
+        #[cfg(all(feature = "tls", not(feature = "rustls")))]
+        let connector = hyper_tls::HttpsConnector::new_with_connector(connector);
+
+        #[cfg(all(feature = "rustls", not(feature = "tls")))]
+        let connector = hyper_rustls::HttpsConnectorBuilder::new()
+            .with_native_roots()
+            .https_or_http()
+            .enable_http2()
+            .wrap_connector(connector);
+
+        #[cfg(all(feature = "rustls", feature = "tls"))]
+        compile_error!(
```

Comment:
It can be very inconvenient in large applications where the code can migrate to rustls gradually or in cases, where different parts of user's code should use different backends for some reason (i.e. I have such an example).

Reply:
Understood, I've removed the compile error in favor of having `rustls` take priority over `tls`.